class-plugin-autoupdate-filter.php

```diff
@@ -36,68 +53,148 @@ public function init() {
+				// Current plugin version to be updated to
+				$version = isset( $item->new_version ) ? $item->new_version : null;
+
+				if ( null === $version ) {
+					continue; // If there's no new version to update to, just continue with the next rule.
+				}
+
+				// Check if the version to update is in the range we should pause
+				if ( isset( $rule['skip_exact_version'] ) && $version === $rule['skip_exact_version'] ) {
+					return false; // Don't update if the exact version should be skipped.
+				}
+				if ( isset( $rule['skip_version_higher_than_or_equal_to'] ) && version_compare( $version, $rule['skip_version_higher_than_or_equal_to'], '>=' ) ) {
+					return false; // Don't update if the version is higher than or equal to the one in the rule.
+				}
+				if ( isset( $rule['skip_version_lower_than'] ) && version_compare( $version, $rule['skip_version_lower_than'], '<' ) ) {
+					return false; // Don't update if the version is lower than the one in the rule.
+				}
+			}
+		}
+
+		return $update; // If none of the rules match, allow updating the plugin.
+	}
+
+	/**
+	 * Enable or disable plugin auto-updates based on time and day of the week.
+	 *
+	 * @param bool   $update Whether to update the plugin or not.
+	 * @param object $item   The plugin or core update object.
+	 *
+	 * @return bool True to update, false to not update.
+	 */
 	public function auto_update_specific_times( $update, $item ) {
+		$now = gmdate( 'Y-m-d H:i:s' );
 
-		$holidays = array(
+		// Apply pre-defined pause periods
+		$pause_periods = array(
 			'christmas' => array(
-				'start' => gmdate( "Y" ) . '-12-23 00:00:00',
-				'end'   => gmdate( "Y" ) . '-12-31 23:59:59',
+				'start' => gmdate( 'Y' ) . '-12-23 00:00:00',
+				'end'   => gmdate( 'Y' ) . '-12-31 23:59:59',
 			),
 			'new_years' => array(
-				'start' => gmdate( "Y" ) . '-01-01 00:00:00',
-				'end'   => gmdate( "Y" ) . '-01-02 23:59:59',
-			),
-			'wait_for_next_woo_release' => array(
-				'start' => '2024-02-16 00:00:00',
-				'end'   => '2024-02-20 23:59:59',
+				'start' => gmdate( 'Y' ) . '-01-01 00:00:00',
+				'end'   => gmdate( 'Y' ) . '-01-02 23:59:59',
 			),
 		);
-		$holidays = apply_filters( 'plugin_autoupdate_filter_holidays', $holidays );
 
-		$now = gmdate( 'Y-m-d H:i:s' );
+		// Add pause periods from JSON settings
+		if ( isset( $this->settings['pause_periods'] ) ) {
+			foreach ( $this->settings['pause_periods'] as $extra_pause_period ) {
+				if ( isset( $extra_pause_period['name'], $extra_pause_period['start'], $extra_pause_period['end'] ) ) {
+					$pause_periods[ $extra_pause_period['name'] ] = array(
+						'start' => $extra_pause_period['start'],
+						'end'   => $extra_pause_period['end'],
+					);
+				}
+			}
+		}
 
-		foreach ( $holidays as $holiday ) {
-			$start = $holiday['start'];
-			$end   = $holiday['end'];
+		$pause_periods = apply_filters( 'plugin_autoupdate_filter_holidays', $pause_periods ); // keep for backward compatibility
```

Comment:
A use-case for https://developer.wordpress.org/reference/functions/apply_filters_deprecated/ maybe? 😄

Reply:
Good tip! Will incorporate this one.